petpal/applications/views.py

```diff
@@ -14,13 +25,110 @@
 PERMISSION:
 SUCCESS:
 '''
+
 @api_view(['GET', 'POST'])
+@permission_classes([IsAuthenticated])
 def applications_list_and_create_view(request):
     if request.method == 'GET':
-        return Response({'msg': 'all applications'}, status=status.HTTP_200_OK)
+        curr_user = PetPalUser.objects.get(pk=request.user.id)
+        status_filter = request.query_params.get('status')
+        sort_option = request.query_params.get('sort')
+        
+        
+        if curr_user.role == PetPalUser.Role.SHELTER:
+            all_apps = Application.objects.filter(shelter=curr_user)
+        elif curr_user.role == PetPalUser.Role.SEEKER:
+            all_apps = Application.objects.filter(seeker=curr_user)
+        else:
+            return Response({'error':f'invalid user role: {curr_user.role}'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+        if status_filter in Application.ALLOWED_STATUS:
+            all_apps = all_apps.filter(status=status_filter)
+
+        if sort_option == 'created_time':
+            all_apps = all_apps.order_by('-created_time')
+        elif sort_option == 'last_updated':
+            all_apps = all_apps.order_by('-last_updated')
+        else:
+            all_apps = all_apps.order_by('-created_time')
+
+
+        paginator = PageNumberPagination()
+        paginator.page_size = 10
+
+        paginated_apps = paginator.paginate_queryset(all_apps, request)
+
+        serialized_apps = ApplicationSerializer(data=paginated_apps, many=True)
+        serialized_apps.is_valid()
+        return JsonResponse({'data': serialized_apps.data}, status=status.HTTP_200_OK)
```

Comment:
change to return paginator.get_paginated_response({'data': serialized_apps.data}) instead of JsonResponse to actually show pages